src/run/run.c

```diff
@@ -799,6 +804,17 @@ static int parse_argv_sudo_mode(int argc, char *argv[]) {
         assert(argc >= 0);
         assert(argv);
 
+        const char *e = secure_getenv("SYSTEMD_RUN_SHELL_PROMPT_PREFIX");
```

Comment:
It's better to parse this from argv first, and fall back to this bit of logic only if we don't get anything?

Reply:
All other opts follow that pattern too